impl/hash_map.py

```diff
@@ -0,0 +1,71 @@
+from typing import Any, NamedTuple, Union
+
+from interface.hash_map import HashMapInterface
+
+
+class Pair(NamedTuple):
+    key: Any
+    value: Any
+
+
+class HashMap(HashMapInterface):
+    def __init__(self, size=8):
+        if size < 1:
+            raise ValueError("Size must be a positive number")
+        self._slots = size * [None]
```

Comment:
- I think would be better to store the initial size and initialize slots by None
- Initialize array the `enter` magic method
```
self._size = size
self._slots = None
```